frontend/pages/shelter/pet/add.tsx

```diff
@@ -0,0 +1,443 @@
+										<Radio value="Sick">Sick</Radio>
+										<Radio value="Fostered">Fostered</Radio>
+										<Radio value="Adopted">Adopted</Radio>
+									</Radio.Group>
+								}
+							/>
+							<DataField
+								required
+								label="Date of Birth"
+								data={
+									<DatePicker
+										onChange={(val) => onDateChange(val as moment.Moment, "dateOfBirth")}
+										defaultValue={moment(pet.dateOfBirth)}
+										format="YYYY/MM/DD"
+										style={{ width: "100%" }}
+									/>
+								}
+							/>
+							<DataField
+								required
+								label="Breed"
+								data={
+									<Select
+										style={{ width: "100%" }}
+										value={pet.breed}
+										onChange={(value: string) => onSelectChange(value, "breed")}>
+										<Select.Option value="persian">Persian</Select.Option>
+										<Select.Option value="mainecoon">Maine Coon</Select.Option>
+										<Select.Option value="bengal">Bengal</Select.Option>
+										<Select.Option value="britishshorthair">British Shorthair</Select.Option>
+										<Select.Option value="siamese">Siamese</Select.Option>
+									</Select>
+								}
+							/>
+							<DataField
+								required
+								label="Fur Color"
+								data={
+									<Select
+										style={{ width: "100%" }}
+										mode="multiple"
+										value={pet.furColor}>
+										<Select.Option value="brown">Brown</Select.Option>
+										<Select.Option value="white">White</Select.Option>
+										<Select.Option value="cgrayat">Gray</Select.Option>
+									</Select>
+								}
+							/>
+
+							<DataField
+								required
+								label="Toilet Trained"
+								data={
+									<Radio.Group
+										value={pet.toiletTrained ? "yes" : "no"}
+										onChange={(e) => onRadioChange(e, "toiletTrained", true)}>
+										<Radio value="yes">Yes</Radio>
+										<Radio value="no">No</Radio>
+									</Radio.Group>
+								}
+								marginBottom={0}
+							/>
+						</div>
+					</Flex>
+				</InnerContent>
+			</Container>
+		</ShelterLayout>
+	);
+}
+
+// =============================================================================
+// Secondary Components
+// =============================================================================
+
+type ImageGalleryProps = {
+	images?: string[];
+	onChange?: (images: string[]) => void;
+};
+
+const ImageGallery = ({ images = [], onChange = () => { } }: ImageGalleryProps) => {
+	const imageUploadRef = useRef<HTMLInputElement>(null);
+	const [pickedImage, setPickedImage] = useState<File | null>(null);
+	const [preview, setPreview] = useState<string>();
+	const [cloned, setCloned] = useState([...images]);
+
+	useEffect(() => {
+		if (!pickedImage) return;
+		const reader = new FileReader();
+		reader.onloadend = () => {
+			setCloned(prev => {
+				return [...prev, reader.result as string];
+			})
+		};
+
+		reader.readAsDataURL(pickedImage);
+	}, [pickedImage]);
+
+	useEffect(() => {
+		if (!onChange) return;
```

Comment:
Following up on @ryhuz comment, `onChange` is always present. `ImageGalleryProps` should not have optional `onChange` as it is always defined.